### PR TITLE
Error handling for add task and add lesson commands

### LINKED ISSUE #59: Handle invalid input when adding lesson

### LINKED ISSUE #60: Handle invalid input when adding task

### PR DESCRIPTION
**Pull request by ivanchongzhien**
Handles most invalid inputs for command to add task and lesson.
Fixed bug in checkIndices method where some indices were skipped and not checked, resulting in out of bounds access.
For checking indices (checkIndices method), non-integer input is automatically removed from the array list. It no longer throws and exception.

Resolves #59, resolves #60

### CODE REVIEW

File: src/main/java/seedu/duke/parser/Parser.java [hunk 1]
```diff
@@ -375,19 +430,22 @@ private LocalDate convertToDate(String string) throws DateTimeParseException {
...
+        
         for (String word : words) {
-            // TODO - inform team using this
-            index = Integer.parseInt(word);
-            rawIndices.add(index);
+            try {
+                index = Integer.parseInt(word);
+                rawIndices.add(index);
+            } catch (NumberFormatException ignored) {
+                // Non-integer inputs are ignored and will not be added to the array list rawIndices.
```
Comment (aliciatay-zls) [inline src/main/java/seedu/duke/parser/Parser.java:444]: Could you print messages for NumberFormatException and ArrayIndexOutOfBounds? I was doing this just now so maybe you could just use these in the Messages class:

`public static final String MESSAGE_INVALID_INDICES = "Please enter numerical indices.";`
`public static final String MESSAGE_OUT_OF_BOUNDS_INDICES = "Please enter an index that is within range.";`

File: src/main/java/seedu/duke/parser/Parser.java [hunk 2]
```diff
@@ -396,10 +454,11 @@ private LocalDate convertToDate(String string) throws DateTimeParseException {
             }
         }
 
-        // remove out of bounds/ invalid index
+        // Remove out of bounds/ invalid index
         for (int i = 0; i < indices.size(); i++) {
-            if (indices.get(i) > max || indices.get(i) < 1) {
+            if (indices.get(i) > max || indices.get(i) <= 0) {
```
Comment (aliciatay-zls) [inline src/main/java/seedu/duke/parser/Parser.java:459]: Right, thanks!

**Review by aliciatay-zls — CHANGES_REQUESTED** [review]
LGTM!

**Review by aliciatay-zls — APPROVED** [review]

**Comment by ivanchongzhien** [comment]
After merging, can @H-horizon and @aliciatay-zls check your respective parts (Hemrish -> add lesson, Alicia -> add task) in the parser. 
There are magic string warnings which I have put there for now as placeholders. (look for the comment "//PLACEHOLDER")
Can you replace them with your own proper warnings by declaring them as constants and simply replacing them in the placeholders?

File: src/main/java/seedu/duke/parser/Parser.java [hunk 3]
```diff
@@ -286,15 +292,20 @@ private ModuleCommands parseInModuleCommandsFromInput(String input) {
...
         String[] allDetails = new String[ENTRY_LESSON_MAX_PARSER];
-        Arrays.fill(allDetails, "");
+        Arrays.fill(allDetails, PLACEHOLDER);
 
         // to remove only the first two words "add lesson"
-        String[] lessonDetails = input.trim().split(" ", 3);
+        String[] lessonDetails = input.trim().split(WHITESPACE, 3);
+
+        // ERROR - User does not enter any parameters.
+        if (lessonDetails.length < 3) {
```
Comment (H-horizon) [inline src/main/java/seedu/duke/parser/Parser.java:305]: Preferable to change the magic number to a constant

**Review by H-horizon — COMMENTED** [review]
Don't forget about the magic numbers and strings. LGTM

File: src/main/java/seedu/duke/parser/Parser.java (hunk 3 above)
Comment (ivanchongzhien) [reply src/main/java/seedu/duke/parser/Parser.java:305]: Magic strings are as placeholders for now.

**Pull request merged**